### DISCUSSION TITLE
Tag pain points

### DISCUSSION THREAD

**Discussion by jsquaredosquared** [freeplane/freeplane#2103] [Help] [answered]
Hi @dpolivaev,

Thanks Freeplane devs for the work on tags :smiley: :+1: . I have been using them for a while and they're great. 

I just have a few issues with searching, renaming, and colouring tags:

## Searching tags

As I type in the name of a tag, the list of tags is filtered to match. But once I have typed in the full name of the tag, the tags are no longer filtered and the full list of tags returns, even before I have had a chance to select the tag I was searching for:

https://github.com/user-attachments/assets/c965ddeb-4c33-4eff-a43e-c81a2e0ecebd

Is this the expected behaviour? If not, is it possible to fix this?

## Renaming tags

Is it possible to rename tags? I was hoping that if I changed the name of the tag in the tag manager, all the instances of that tag would be renamed as well:

https://github.com/user-attachments/assets/4d6f064b-d434-4f95-a1c3-25a36f61df60

## Randomizing tag colour

This isn't a huge issue, but sometimes when I create tags some of them will get the same colour. I don't want to have to think about which colour to change to, so would it be possible to implement a button I could click to randomize the colour of the tag? Totally understand if this is deemed unnecessary, but just thought I'd mention it anyway.

Any help would be appreciated.
Thanks :D

**Comment by dpolivaev** [comment, marked as answer]
I think that all problems you report have already been resolved, and can not recommend you enough to always try latest stable and latest preview version before reporting any bugs. In your video i can even see that information about available update is shown in the status line, but you seem to ignore it. How can you get the bugs fixed if you do not download versions containing the bug fixes?

And if you want to report any unexpected behavior, please do it by opening an issue and not by starting a discussion because discussions are for the community, and issues labeled as "bugs" are for me to resolve.

In such cases I always ask to attach the mind map files mentioned in the bug report, only screenshots or videos are generally not sufficient because I can not use them to easily reproduce the reported issue.

Regards

**Reply by jsquaredosquared** [reply]
😢 Sorry. I didn't know about such rules. I will take note for next time.
